wagtail/admin/menu.py

```diff
@@ -11,10 +11,11 @@
 class MenuItem(metaclass=MediaDefiningClass):
     template = 'wagtailadmin/shared/menu_item.html'
 
-    def __init__(self, label, url, name=None, classnames='', attrs=None, order=1000):
+    def __init__(self, label, url, name=None, classnames='', icon_name='', attrs=None, order=1000):
         self.label = label
         self.url = url
         self.classnames = classnames
+        self.icon_name = icon_name
```

Comment:
Is `classnames` used for any other purpose than icons? If not we should remove it, and-or have a deprecation path then remove it.

Reply:
Two things:
1. In order to maintain compatibility for wagtailfontawesome, we'd have to do something like, if a name is passed to `menu_icon` that begins with `fa-` ([see example in bakerydemo](https://github.com/wagtail/bakerydemo/blob/99b65a5e24038e9eadcd8a3d5198efd827ab1952/bakerydemo/base/wagtail_hooks.py#L40)), then instead of using our `{% icon %}` tag, output the necessary classes. Not a huge deal, but… 
2. It is a documented API for customizing new menu items, generally: https://docs.wagtail.io/en/v2.5.1/reference/hooks.html?highlight=menuitem#register-admin-menu-item. Though it only mentions using it to add icons, I am sure there are people out there using it for more than that. We could create a deprecation path, but I'm not sure if we want to take away the ability to add custom classes to new menu items.